src/main/java/org/commcare/formplayer/services/MenuSessionFactory.java

```diff
@@ -101,11 +106,15 @@ public void rebuildSessionFromFrame(MenuSession menuSession, CaseSearchHelper ca
                     for (StackFrameStep step : steps) {
                         if (step.getId().equals(options[i].getCommandID())) {
                             currentStep = String.valueOf(i);
+                            processedSteps.add(step);
```

Comment:
I notice this loop has no break statement in it. I think it should since I can't think of a scenario where more than one step would match.

Reply:
I was thinking that too. I will make that change in this separate [PR](https://github.com/dimagi/formplayer/pull/1562)

Aside note: I was speaking to Shubham about commit [4ef0364](https://github.com/dimagi/formplayer/pull/1560/commits/4ef0364b1a911e25fd90f45a6a4615e3d15f7d96) and he suggested doing this in a separate PR with more scrutiny (doing QA). So may as well have this extra check done on the `break` statement too.

Reply:
In that case, should this line be moved out of the for loop since only the last matching step would be 'processed'. I don't think it should be possible for multiple steps to match but I can't prove that.